scipy/spatial/_geometric_slerp.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import division, print_function, absolute_import
+
+__all__ = ['geometric_slerp']
+
+import math
+import numpy as np
+from . import _slerp
+from scipy.spatial.distance import euclidean
+
+def geometric_slerp(start,
+                    end,
+                    t,
+                    tol=1e-7):
+    """
+    Spherical linear interpolation along unit-radius
+    great circle arc in arbitrary dimensional space.
+
+    Parameters
+    ----------
+    start : (D, ) array-like
+        Single D-dimensional input coordinate in a 1-D array-like
+        object.
+        Starting coordinate for interpolation will be converted
+        to float64.
+    end : (D, ) array-like
+        Single D-dimensional input coordinate in a 1-D array-like
+        object.
+        End coordinate for interpolation will be converted to
+        float64.
+    t: array
+        An array of doubles representing interpolation parameters,
+        with values required in the inclusive interval between
+        0 and 1. A common approach is to generate the array with
+        ``np.linspace(0, 1, n_pts)`` for linearly spaced points.
+        Ascending, descending, and scrambled orders are permitted.
+    tol: float
```

Comment:
From the first look I'm not convinced that this parameter and the corresponding checking is necessary for this function. Seems not very important.

Reply:
I don't get it--how are your proposing that antipodal points are handled? That's a case of degenerate input, and allowing the user to control the sensitivity in detecting that degeneracy seems sensible to me.

Reply:
I see now that for antipodes division by zero may occur, so some guards are necessary. My bad.

However I see that the code posted by @pmla does not contain such division and according to his benchmarks work more stable for angles closes to `pi`. (I don't understand how it works though.)

I'd say that if we can avoid numerical instability then we should definitely do so. And in this case I would advocate for removing `tol`. Even if we have the situation with truly ambiguous interpolation curve --- it is still a valid curve and I believe a user should decide separately whether he wants such situation or not.

Reply:
Although I've removed the Cython code & used Peter's implementation, there were stability tradeoffs elsewhere (reproducing values exactly at 0) noted in my commit messages. 

I'm still maintaining that the infinite possible ambiguous paths between antipodes should raise rather than potentially returning one of those possible values.

Reply:
I had a look into the stability issues. It turns out that they result from the tiny errors in `np.sin` and `np.cos`. The previous implementation used only `np.sin`, whereas the current one uses `np.cos` and `np.sin`. Since one of the test inputs is `t=1/3` we have:
```
>>> np.cos(np.pi / 3) - 0.5
1.1102230246251565e-16
>>> np.sin(np.pi / 3) - np.sqrt(3) / 2
0
```
The use of `np.cos` means we cannot reproduce 0 exactly.

However, if a test input were `t=1/4` we would have:
```
>>> np.cos(np.pi / 4) - np.sqrt(2) / 2
0.0
>>> np.sin(np.pi / 4) - np.sqrt(2) / 2
-1.1102230246251565e-16
```
Now the use of `np.sin` means we cannot reproduce 0 exactly.

In other words, the ability of the previous implementation to reproduce values at 0 depends on the choice of input.

Reply:
The stability was not clearly superior in all aspects for the new implementation, but I can live with that tradeoff if it makes reviewers happy.

The antipodal discussion is happening in two places now, but the plot/analysis below should be clear on why I'm pushing for an exception over a warning once a user is close to antipodal input. We could push the threshold to the absolute limit, but if the algorithm changes under the hood the pathological threshold may change anyway---if you're getting close to antipodes you're probably doing something wrong and should consider just using the function twice & splitting up your workflow to get normal paths that may be fused.